packages/contracts-bedrock/scripts/DeployOPChain.s.sol

```diff
@@ -72,6 +80,16 @@ contract DeployOPChainInput is BaseDeployIO {
             _l2ChainId = _value;
         } else if (_sel == this.gasLimit.selector) {
             _gasLimit = SafeCast.toUint64(_value);
+        } else if (_sel == this.disputeGameType.selector) {
+            _disputeGameType = GameType.wrap(SafeCast.toUint32(_value));
```

Comment:
@Inphi There are only a few supported game type integers right? 0, 1, and 2^32 -1 (alphabet VM) IIRC? Where are the allowed types documented?

Reply:
It depends on exactly what we mean by supported.  The `DisputeGameFactory` is deliberately setup like a plugin system so it supports registering an implementation for any game type. The types op-challenger knows about are https://github.com/ethereum-optimism/optimism/blob/52336b43a6ba72990383f4baa6a7d6619731ef5d/op-challenger/game/fault/types/types.go#L27-L33

`FastGameType` and `AlphabetGameType` are only used for testing/local devnet.  We may never deploy `AsteriscGameType` since that's asterisc/op-program combination and we're likely to only actually ship cannon/op-program and asterisc/kona.

But non-standard chains could easily add their own game type - eg a ZK proof would be a new game type.